tests/common/test_keyword_checklist.py

```diff
@@ -31,3 +38,41 @@ def test_keyword_set_n(self):
                 f.write(text)
                 f.write('\n')
         self.assertFalse(wrong_items, "Keywords list has been rearranged and updated")
+
+    def test_morpheme_set_n(self):
+        # check whether the morphemes are optimized and updates the file with optimized list
+        morpheme_checklist_data = Util.read_data(KeywordChecklist.MORPHEME_PATH)
+        original_morpheme_list = morpheme_checklist_data.decode().split()
+        original_morpheme_list_len = len(original_morpheme_list)
+        morpheme_set = set(original_morpheme_list)
+        optimized_morpheme_list = sorted(list(morpheme_set), key=lambda e: len(e))
+
+        optimized = False
+        while not optimized:
+            to_remove = set()
+            l = len(optimized_morpheme_list)
+            n = 0
+            for i in optimized_morpheme_list:
+                n += 1
+                if i.lower() == i and 3 <= len(i):
+                    # search in rest list whether the items have the substring
+                    m = n
+                    while m < l:
+                        if i in optimized_morpheme_list[m]:
+                            # should be removed
+                            to_remove.add(optimized_morpheme_list[m])
+                        m += 1
+                else:
+                    # wrong - must be removed
+                    to_remove.add(i)
+            for x in to_remove:
+                optimized_morpheme_list.remove(x)
+            optimized = 0 == len(to_remove)
```

Comment:
What about change variables like below?
They are hard to understand at once. :dizzy_face: 

```suggestion
            to_remove_set = set()
            optimized_morpheme_list_len = len(optimized_morpheme_list)
            idx = 0
            for optimized_morpheme in optimized_morpheme_list:
                idx += 1
                if optimized_morpheme.lower() == optimized_morpheme and 3 <= len(optimized_morpheme):
                    # search in rest list whether the items have the substring
                    local_idx = idx
                    while local_idx < optimized_morpheme_list_len:
                        if optimized_morpheme in optimized_morpheme_list[local_idx]:
                            # should be removed
                            to_remove.add(optimized_morpheme_list[local_idx])
                        local_idx += 1
                else:
                    # wrong - must be removed
                    to_remove_set.add(optimized_morpheme)
            for to_remove in to_remove_set:
                optimized_morpheme_list.remove(to_remove)
```

Reply:
done

Reply:
Thank you for a quick fix.
I've checked this with 4b28845.